mailbox/backup/src/test/java/org/apache/james/mailbox/backup/DefaultMailboxBackupTest.java

```diff
@@ -0,0 +1,148 @@
+            IOUtils.copy(content, out);
+        }
+    }
+
+    private void createMailBoxWithMessage(MailboxSession session, MailboxPath mailboxPath, MailboxMessage... messages) throws Exception {
+        MailboxId mailboxId = mailboxManager.createMailbox(mailboxPath, session).get();
+        Arrays.stream(messages).forEach(Throwing.consumer(message ->
+                mailboxManager.getMailbox(mailboxId, session).appendMessage(MessageManager.AppendCommand.from(message.getFullContent()), session)
+            )
+        );
+    }
+
+    private ZipAssert assertThatZipFromIdContainsOnly(Publisher<BlobId> blobIdPublisher, EntryChecks... onlyEntriesMatching) throws Exception {
+        BlobId blobId = Mono.from(blobIdPublisher).block();
+        readFromStoreAndCopyInFile(blobId);
+
+        try (ZipFile zipFile = new ZipFile(destination)) {
+            return ZipAssert.assertThatZip(zipFile).containsOnlyEntriesMatching(onlyEntriesMatching);
+        }
+    }
+
+    @Test
+    void doBackupWithoutMailboxShouldStoreEmptyBackup() throws Exception {
+        Publisher<BlobId> res = backup.backupAccount(USER1);
+
+        assertThatZipFromIdContainsOnly(res);
+    }
+
+    @Test
+    void doBackupWithoutMessageShouldStoreAnArchiveWithOnlyOneEntry() throws Exception {
+        MailboxSession session = mailboxManager.createSystemSession(ManagerTestResources.USER);
+        createMailBoxWithMessage(session, MAILBOX_PATH_USER1_MAILBOX1);
+
+        Publisher<BlobId> res = backup.backupAccount(USER1);
+
+        assertThatZipFromIdContainsOnly(res, EntryChecks.hasName(MAILBOX_1_NAME + "/").isDirectory());
+    }
+
+    @Test
+    void doBackupWithOneMessageShouldStoreAnArchiveWithTwoEntries() throws Exception {
+        MailboxSession session = mailboxManager.createSystemSession(ManagerTestResources.USER);
+        createMailBoxWithMessage(session, MAILBOX_PATH_USER1_MAILBOX1, MESSAGE_1);
+
+        Publisher<BlobId> res = backup.backupAccount(USER1);
+
+        assertThatZipFromIdContainsOnly(res,
+            EntryChecks.hasName(MAILBOX_1_NAME + "/").isDirectory(),
+            EntryChecks.hasName(MESSAGE_ID_1.serialize()).hasStringContent(MESSAGE_CONTENT_1)
+        );
+    }
+
+    @Test
+    void doBackupShouldOnlyArchiveTheMailboxOfTheUser() throws Exception {
+        MailboxSession session = mailboxManager.createSystemSession(ManagerTestResources.USER);
+        MailboxSession otherSession = mailboxManager.createSystemSession(ManagerTestResources.OTHER_USER);
+
+        createMailBoxWithMessage(session, MAILBOX_PATH_USER1_MAILBOX1, MESSAGE_1);
+        createMailBoxWithMessage(otherSession, MAILBOX_PATH_OTHER_USER_MAILBOX1, MESSAGE_1_OTHER_USER);
+
+        Publisher<BlobId> res = backup.backupAccount(USER1);
+
+        assertThatZipFromIdContainsOnly(res,
+            EntryChecks.hasName(MAILBOX_1_NAME + "/").isDirectory(),
+            EntryChecks.hasName(MESSAGE_ID_1.serialize()).hasStringContent(MESSAGE_CONTENT_1)
+        );
+    }
+}
```

Comment:
Should we have a test with a backup of an user with multiple mailboxes and messages as well?